File: DevExpress-Demo/DevExpress.ProductsDemo.Wpf/CS/DevExpress.ProductsDemo.Wpf/Controls/PhotosGallery/ViewTypeToBoolConverter.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using System.Windows.Data;

namespace ProductsDemo {
    public class ViewTypeToBoolConverter : IValueConverter {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture) {
            if(targetType == typeof(bool) && value is PhotoGalleryViewType && parameter is PhotoGalleryViewType)
                return (PhotoGalleryViewType)value == (PhotoGalleryViewType)parameter;
            return false;
        }
        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) {
            if(value is bool)
                if(targetType == typeof(PhotoGalleryViewType)) {
                    return (bool)value ? PhotoGalleryViewType.Gallery : PhotoGalleryViewType.Map;
                }
            return null;
        }
    }
    public class ViewTypeToVisibilityConverter : IValueConverter {
        object IValueConverter.Convert(object value, Type targetType, object parameter, CultureInfo culture) {
            if (targetType == typeof(Visibility) && value is PhotoGalleryViewType && parameter is PhotoGalleryViewType)
                return (PhotoGalleryViewType)value == (PhotoGalleryViewType)parameter ? Visibility.Visible : Visibility.Hidden;
            return Visibility.Hidden;
        }
        object IValueConverter.ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) {
            if (value is Visibility)
                if (targetType == typeof(PhotoGalleryViewType)) {
                    return (Visibility)value == Visibility.Visible ? PhotoGalleryViewType.Gallery : PhotoGalleryViewType.Map;
                }
            return null;
        }
    }
    public enum PhotoGalleryViewType {
        Map,
        Gallery,
        Detail
    }
}

File: DevExpress-Demo/DevExpress.ProductsDemo.Wpf/VB/DevExpress.ProductsDemo.Wpf/Controls/PhotosGallery/ViewTypeToBoolConverter.vb
Imports System
Imports System.Globalization
Imports System.Windows
Imports System.Windows.Data

Namespace ProductsDemo

    Public Class ViewTypeToBoolConverter
        Implements IValueConverter

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
            If targetType Is GetType(Boolean) AndAlso TypeOf value Is PhotoGalleryViewType AndAlso TypeOf parameter Is PhotoGalleryViewType Then Return CType(value, PhotoGalleryViewType) = CType(parameter, PhotoGalleryViewType)
            Return False
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            If TypeOf value Is Boolean Then
                If targetType Is GetType(PhotoGalleryViewType) Then
                    Return If(CBool(value), PhotoGalleryViewType.Gallery, PhotoGalleryViewType.Map)
                End If
            End If

            Return Nothing
        End Function
    End Class

    Public Class ViewTypeToVisibilityConverter
        Implements IValueConverter

        Private Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
            If targetType Is GetType(Visibility) AndAlso TypeOf value Is PhotoGalleryViewType AndAlso TypeOf parameter Is PhotoGalleryViewType Then Return If(CType(value, PhotoGalleryViewType) = CType(parameter, PhotoGalleryViewType), Visibility.Visible, Visibility.Hidden)
            Return Visibility.Hidden
        End Function

        Private Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            If TypeOf value Is Visibility Then
                If targetType Is GetType(PhotoGalleryViewType) Then
                    Return If(CType(value, Visibility) = Visibility.Visible, PhotoGalleryViewType.Gallery, PhotoGalleryViewType.Map)
                End If
            End If

            Return Nothing
        End Function
    End Class

    Public Enum PhotoGalleryViewType
        Map
        Gallery
        Detail
    End Enum
End Namespace
